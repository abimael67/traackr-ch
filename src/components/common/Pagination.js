const Pagination = ({ nextPage, prevPage, gotoPage, currentPage, pagesCount }) => {
    return (
        <nav className="pagination is-centered" role="navigation" aria-label="pagination">
            <a className="pagination-previous" onClick={prevPage}>Previous</a>
            <a className="pagination-next" onClick={nextPage}>Next page</a>
            <ul className="pagination-list">
                {
                    [...Array(pagesCount)].map((_, pageInd) => <li key={pageInd} >
                        <a className={`pagination-link ${currentPage === pageInd ? 'is-current' : ''}`} aria-label={`Goto page ${pageInd + 1}`} data-testid="pagination-link" onClick={() => gotoPage(pageInd)}>{pageInd + 1}</a>
                    </li>
                    )
                }
            </ul>
        </nav>
    )
}

export default Pagination