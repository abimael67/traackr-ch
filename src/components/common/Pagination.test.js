import { render, screen } from '@testing-library/react'
import Pagination from './Pagination'
import renderer from 'react-test-renderer'

describe('pagination tests', () => {
  it('renders the right number of page links', () => {
    render(<Pagination pagesCount={10} />);
    const links = screen.getAllByTestId('pagination-link');
    expect(links).toHaveLength(10);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Pagination pagesCount={3} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

