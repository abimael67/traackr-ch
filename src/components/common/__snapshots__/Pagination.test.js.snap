// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination tests renders correctly 1`] = `
<nav
  aria-label="pagination"
  className="pagination is-centered"
  role="navigation"
>
  <a
    className="pagination-previous"
  >
    Previous
  </a>
  <a
    className="pagination-next"
  >
    Next page
  </a>
  <ul
    className="pagination-list"
  >
    <li>
      <a
        aria-label="Goto page 1"
        className="pagination-link "
        data-testid="pagination-link"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li>
      <a
        aria-label="Goto page 2"
        className="pagination-link "
        data-testid="pagination-link"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li>
      <a
        aria-label="Goto page 3"
        className="pagination-link "
        data-testid="pagination-link"
        onClick={[Function]}
      >
        3
      </a>
    </li>
  </ul>
</nav>
`;
